src/pip/_internal/index.py

```diff
@@ -254,6 +255,63 @@ def _get_html_page(link, session=None):
     return None
 
 
+class FoundCandidates(object):
+    """A collection of candidates, returned by `PackageFinder.find_candidates`.
+
+    Arguments:
+
+    * `candidates`: A sequence of all available candidates found.
+    * `specifier`: Specifier to filter applicable versions.
+    * `prereleases`: Whether prereleases should be accounted. Pass None to
+        infer from the specifier.
+    * `sort_key`: A callable used as the key function when choosing the best
+        candidate.
+    """
+    def __init__(
+        self,
+        candidates,     # type: List[InstallationCandidate]
+        specifier,      # type: specifiers.BaseSpecifier
```

Comment:
Again, I think specifier should be optional (defaulting to the empty `SpecifierSet`).

Reply:
I made the `specifier` argument optional in `finder.find_candidates()`, but kept it required in `FoundCandidates` constructor.